src/shared/Core/Interop/Linux/LinuxFileSystem.cs

```diff
@@ -8,8 +8,18 @@ public class LinuxFileSystem : PosixFileSystem
     {
         public override bool IsSamePath(string a, string b)
         {
-            a = Path.GetFileName(a);
-            b = Path.GetFileName(b);
+            if (string.IsNullOrWhiteSpace(a) || string.IsNullOrWhiteSpace(b))
+            {
+                return false;
+            }
+
+            // Normalize paths
+            a = Path.GetFullPath(a);
+            b = Path.GetFullPath(b);
+
+            // Resolve symbolic links
+            a = ResolveSymbolicLink(a);
```

Comment:
Does this resolve symlinks recursively? The singular in the function name suggests that it handles only one iteration, i.e. the result of `ResolveSymbolicLink()` might point to _another_ symlink...

Reply:
The implementation currently does indeed resolve _all_ links (up to some hardcoded max in the .NET libraries.. like 40 for POSIX and 60 for Windows).

I can rename this to `ResolveSymbolicLinks` (to make it a bit clearer) :-)

Reply:
> I can rename this to `ResolveSymbolicLinks` (to make it a bit clearer) :-)

I would like that ;-)

Reply:
[](url)